src/enumerable.rb

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    length.times do |i|
+      yield(self[i])
+    end
+  end
+
+  def my_each_with_index
+    length.times do |i|
+      yield(self[i], i)
+    end
+  end
+
+  def my_select
+    selected_items = []
+    my_each do |item|
+      selected_items << item if yield(item)
+    end
+    selected_items
+  end
+
+  def my_all?
+    my_each { |item| return false unless yield(item) }
+    true
+  end
+
+  def my_any?
+    my_each { |item| return true if yield(item) }
+    false
+  end
+
+  def my_none?
+    my_each { |item| return false if yield(item) }
+    true
+  end
+
+  def my_count
+    count = 0
+    my_each { |item| count += 1 if yield(item) }
```

Comment:
#### 16.  Should return the number of elements in a collection if no block is given:

```
irb(main):076:0> [1,2,3].count
=> 3
```

#### 17.  If we pass an argument with a pattern, it should count the number of times that such given pattern appears in the collection, example:

```
irb(main):084:0> [1,2,1,1,3,4,5,6,67,7,7,4,2,342,34,34,432,14].count(1)
=> 3
```

#### 18. it should return an enumerator when no blocks or arguments are given